### ISSUE TITLE
clientconfig: Support for secure.yaml 

### ISSUE THREAD

**Issue by simonre** [gophercloud/utils#41]
According to this: https://docs.openstack.org/os-client-config/latest/user/configuration.html Openstack supports a secure.yaml file to specify secrets in a different file. I think it makes sense to include it here as well.

**Comment by jtopjian** [comment]
This coincidentally came up in a Terraform issue today :)

**Comment by simonre** [comment]
@jtopjian I'll add this after making sure that clouds-public.yaml is merged correctly in #40 as I think this is very similar.

**Comment by haxorof** [comment]
References:
https://docs.openstack.org/os-client-config/latest/user/configuration.html#splitting-secrets
https://github.com/openstack/openstacksdk/blob/stable/queens/openstack/config/loader.py

**Comment by simonre** [comment]
@jtopjian @haxorof It seems to me that secure.yaml overrides clouds.yaml and clouds-public.yaml overrides both. This does seem a bit unintuitive to me. Is that assessment correct?

**Comment by haxorof** [comment]
@simonre Yes, basically that is very similar to what I interpret it. Looking at the code it seems to me that the secure.yaml can override everything that is defined in clouds.yaml. And when that is done the clouds-public.yaml only overrides the auth part. 

Merge secure.yaml structure into clouds.yaml structure during construction of OpenStackConfig
https://github.com/openstack/openstacksdk/blob/stable/queens/openstack/config/loader.py#L214-L219

Subsequent calls here will update the auth part with what is defined in clouds-public.yaml if some profile is referenced there:
https://github.com/openstack/openstacksdk/blob/stable/queens/openstack/config/loader.py#L1254

@jtopjian What is your interpretation of this?

**Comment by jtopjian** [comment]
Yes, that sounds correct to me. One other are of code which is relevant is:

https://github.com/openstack/openstacksdk/blob/stable/queens/openstack/config/loader.py#L121-L146

I agree this all seems backwards.

I suppose one way of being absolutely sure is to test this with the three files (`clouds.yaml`, `clouds-public.yaml`, and `secure.yaml`) and run a command like

```
$ openstack server list --os-cloud mycloud
```

**Comment by haxorof** [comment]
Now I did some testing and was a bit surprised over the result. Here is a bullet list of what I found:

- A cloud can be defined in `secure.yaml` without any entry in `clouds.yaml`.
- Not valid to only define the cloud key in `clouds.yaml` even if the rest is defined in `secure.yaml`.
- `secure.yaml` override all values in `clouds.yaml`.
- `clouds.yaml` override all values in `clouds-public.yaml`.
- All values are used from `clouds-public.yaml` except profile.
- Possible to define a `clouds.yaml` which only have a cloud which reference a profile with all the values in it.

The logic the above bullets show seems reasonable to me but not what I saw in the source code of the latest version (or I really got confused when looking at it). I think I must check the source code again on exactly this version. 😄

Versions used:
```
openstacksdk 0.11.3
openstack 3.14.0 (Found in queens repo)
```

**Comment by jtopjian** [comment]
Thanks so much for the in-depth testing :)

> or I really got confused when looking at it

That would make two of us.

**Comment by simonre** [comment]
@haxorof Thanks for the testing 👍  
So from what I understand now:
clouds-public.yaml gets overwritten by clouds.yaml which then gets overwritten by secure.yaml
(with the restrictions you saw in your tests).
I think this means that the implementation at the moment isn't quite right in that regard. @jtopjian Is it okay if I fix this in the PR for this issue as well or do you want to open a separate issue/PR for this?

Also 

>> or I really got confused when looking at it
>
> That would make two of us.

Make that three of us

**Comment by jtopjian** [comment]
> okay if I fix this in the PR for this issue as well or do you want to open a separate issue/PR for this?

A single PR is fine, especially since all of the code will go together.

One other thing I'm curious about, and I can do this when I have some time, is to confirm we're looking at the right areas of code. One way of doing this is to add some "print" or "log" checkpoint statements in the code to narrow the various areas down.

It's possible that the reason we're all confused is because we're looking in the wrong areas :smile:

**Comment by jtopjian** [comment]
Implemented in #58
